src/stores/apiStore.ts

```diff
@@ -0,0 +1,149 @@
+import { defineStore, } from "pinia";
+import axios, { type AxiosResponse } from "axios";
+import _ from "lodash";
+import type {
+  BotDetectorApiResponse,
+  BotDetectorApiStore,
+  BotDetectorApiPredictionBreakdownResponse,
+  PatreonApiStore,
+  RuneLiteApiRespose,
+  RuneLiteStore,
+} from "@/interfaces/ApiInterface";
+import type{ IConfigStore } from "@/interfaces/ConfigInterface";
+import { replaceUnderscoreWithSpace, toPercentage } from "@/utils";
+import { useConfigStore } from "@/stores/configStore";
+
+let configStore: undefined | IConfigStore;
+
+// see https://docs.patreon.com/?javascript=#get-api-oauth2-v2-campaigns-campaign_id-members=
+export const usePatreonStore = defineStore({
+  id: "patreonApiStore",
+  state: () =>
+    ({
+      members: [],
+    } as PatreonApiStore),
+  getters: {
+    getMembersFromTier: (state) =>
+      state.members.filter((member) => member.tier),
+  },
+  actions: {},
+});
+
+export const useRuneLiteStore = defineStore({
+  id: "runeLiteApiStore",
+  state: () =>
+    ({
+      totalInstalls: 0,
+    } as RuneLiteStore),
+  getters: {
+    getTotalInstalls: (state) =>
+      state.totalInstalls === 0 ? "Failed getting stats." : state.totalInstalls,
+  },
+  actions: {
+    setProjectStats: function (response: AxiosResponse<RuneLiteApiRespose>) {
+      this.totalInstalls = response.data["bot-detector"];
+    },
+    getProjectStats: function () {
+      if(!configStore) {
+        configStore = useConfigStore();
+        configStore.getApi();
+      }
+
+      const API_URL = configStore.api.runeLite;
+      if (!API_URL) return console.error("Could not load runelite api url from config", configStore.api.runeLite)
+      
+      axios
+        .get(API_URL)
+        .then((response) => this.setProjectStats(response))
+        .catch((error) => console.error(error));
+    },
+  },
+});
+
+export const useBotDetectorApiStore = defineStore({
+  id: "botDetectorApiStore",
+  state: () =>
+    ({
+      totalBans: 0,
+      totalAccounts: 0,
+      totalPlayers: 0,
+      selectedRSN: "",
+      responseStatus: 0,
+      isAwaitingResponse: false,
+      playerName: "",
+      responseData: {} as BotDetectorApiPredictionBreakdownResponse,
+      botDetectorApiUrl: ""
+    } as BotDetectorApiStore),
+  getters: {
+    isReponseStatusNotFound: (state) => state.responseStatus === 404,
+    isResponseStatusOk: (state) => state.responseStatus === 200,
+    orderdPredictions: (state) => {
+      return _(state.responseData.predictions_breakdown)
```

Comment:
This maybe isn't the best way to do this, instead of importing all of `lodash`  i suggest using named imports alongside `flow` [link here](https://lodash.com/docs/4.17.15#flow) which will give a smaller bundle size and it could also be worth looking into `lodash FP` which pairs nicely when using functional programming. [Link here](https://github.com/lodash/lodash/wiki/FP-Guide)

Reply:
I agree on not fully importing lodash, I am however not familiar enough with lodash. I am not sure how to implement you suggestion.

Reply:
I can open a pull request in a few days if you'd like?

Reply:
If the PR is still open sure go ahead :) you can even work on this PR if you want

Reply:
Oh nice, thankyou very much ill take a look at it when I get some spare time from work